typescript/src/actions/solana/transfer.ts

```diff
@@ -0,0 +1,248 @@
+  /**
+   * The token to transfer, or mint address of the SPL token to transfer.
+   */
+  token: "sol" | "usdc" | string;
+  /**
+   * The network to use which will be used to create a Connection, otherwise a Connection can be provided.
+   */
+  network: Network | Connection;
+}
+
+export interface TransferResult {
+  /**
+   * The signature result of the transaction
+   */
+  signature: string;
+}
+
+/**
+ * Transfer SOL or SPL tokens between accounts
+ *
+ * @param apiClient - The API client to use
+ * @param options - The transfer options
+ *
+ * @returns The transfer result
+ */
+export async function transfer(
+  apiClient: CdpOpenApiClientType,
+  options: TransferOptions,
+): Promise<TransferResult> {
+  const connection = getOrCreateConnection({ networkOrConnection: options.network });
+
+  const tx =
+    options.token === "sol"
+      ? await getNativeTransferInstructions({
+          connection,
+          from: options.from,
+          to: options.to,
+          amount: options.amount,
+        })
+      : await getSplTransferInstructions({
+          connection,
+          from: options.from,
+          to: options.to,
+          mintAddress:
+            options.token === "usdc"
+              ? getUsdcMintAddress(await getConnectedNetwork(connection))
+              : options.token,
+          amount: options.amount,
+        });
+
+  const serializedTx = Buffer.from(tx.serialize()).toString("base64");
+
+  const signedTxResponse = await apiClient.signSolanaTransaction(options.from, {
+    transaction: serializedTx,
+  });
+
+  const decodedSignedTx = Buffer.from(signedTxResponse.signedTransaction, "base64");
+
+  const signature = await connection.sendRawTransaction(decodedSignedTx, {
+    skipPreflight: false,
+    maxRetries: 3,
+  });
+
+  return { signature };
+}
+
+type GetNativeTransferInstructionsOptions = Omit<TransferOptions, "token" | "network"> & {
+  /**
+   * The connection to the Solana network
+   */
+  connection: Connection;
+};
+
+/**
+ * Get the instructions for a native SOL transfer
+ *
+ * @param options - The options for the native SOL transfer
+ *
+ * @param options.connection - The Solana connection
+ * @param options.from - The source address
+ * @param options.to - The destination address
+ * @param options.amount - The amount to transfer
+ *
+ * @returns The instructions for the native SOL transfer
+ */
+async function getNativeTransferInstructions({
+  connection,
+  from,
+  to,
+  amount,
+}: GetNativeTransferInstructionsOptions) {
+  const { blockhash } = await connection.getLatestBlockhash();
+
+  const instructions = [
+    SystemProgram.transfer({
+      fromPubkey: new PublicKey(from),
+      toPubkey: new PublicKey(to),
+      lamports: getLamports(amount),
+    }),
+  ];
+
+  const messageV0 = new TransactionMessage({
+    payerKey: new PublicKey(from),
+    recentBlockhash: blockhash,
+    instructions,
+  }).compileToV0Message();
+
+  return new VersionedTransaction(messageV0);
```

Comment:
the function name and docstring says that it returns instructions, but in fact it returns a transaction